client/mysqlbinlog.cc

```diff
@@ -3138,6 +3133,15 @@ int main(int argc, char** argv)
     warning("Did not reach stop datetime '%s' before end of input",
             stop_datetime_str);
 
+  /*
+    Emit a warning in the event that we finished processing input
+    before reaching the boundary indicated by --stop-position.
+  */
+  if ((static_cast<longlong>(stop_position) != stop_position_default) &&
+      stop_position > last_processed_position)
+    warning("Did not reach stop position %llu before end of input",
+            stop_position);
```

Comment:
Funny. It used to check/warn datetime after position, but now it’s before. 😀
No need to change it back though. Positions are “absolute” while datetime (which can go out-of-order) are “relative”, so it feels intuitive to have that last.